### ISSUE TITLE
Is there a reason why all the latest releases are behind the "pre-release" flag?

### ISSUE THREAD

**Issue by randy-halim** [Exiled-Team/EXILED#403]
It took me one month (and some angry friends) before I realized that all the latest releases for EXILED are labeled as `pre-release` - meaning that the installer didn't update by default (thinking that it was already up to date). This led to the overall server breaking for me and having to reconstruct it.

**Comment by PintTheDragon** [comment]
No, not really, but I believe we're migrating to releases instead of pre-releases.

Releases should always work. Pre-releases might not work and may be buggy, but will be the latest versions.

The latter for our pre-releases holds true, but not all all for releases. Currently, the latest releases end up being broken and outdated, while the pre-releases are fully tested versions to go along with SL's releases.

So no, there is no good reason that the latest releases are being marked as pre-releases.

**Comment by galaxy119** [comment]
Actually yes, there is. Ignore Pint he's a monkey.

The reason being - Releases are fully tested, stable and there are no major bugs present. When we add a new feature, or change something that will break plugins, we tag it as a 'pre-release' until those changes/additions have been fully tested and we can consider them stable. This continues for some time, as more people add/change more things. 

Eventually, we will mark a specific version as a release-candidate, everything in it will be included in the next release, and any more changes/etc made to the dev branch after that will not.

This is a very standard and well practiced release schedule many programs use.

Northwood makes this difficult by issuing game updates that require us to make changes to EXILED without any prior notice. This means we *have* to make changes, even if we are currently looking at a release-candidate. Effectively making us go back and re-test everything. 

What's happened this time is, everytime we select a release-candidate, Northwood has made some kind of update that's caused us to need to make changes, and thus eliminated the release candidate from being considered "stable". Thus we run on pre-releases.

We will be addressing this in the very near future, by forcing a possibly 'unstable' build to be the "full release" when there's a game update that breaks EXILED compatability.

**Comment by PintTheDragon** [comment]
> This is a very standard and well practiced release schedule many programs use.
>
> Northwood makes this difficult by issuing game updates that require us to make changes to EXILED without any prior notice. This means we have to make changes, even if we are currently looking at a release-candidate. Effectively making us go back and re-test everything.

That is exactly why there is no good reason to make everything a pre-release. EXILED is forced to follow SL's release schedule, so it doesn't make all that much sense for it to use a contrasting release schedule. Pre-releases can be thought of as beta releases, and while beta releases are more ahead of full releases (and more buggy), that does not make the full release non-functional. The issue with EXILED is that is not able to guarantee that the latest release is still functional, but continues to use a release schedule that assumes it to be. EXILED not being able to have significant testing is a huge issue, but does not justify the current release schedule.

EXILED should create pre-releases for new EXILED features or for beta versions of the game, but the latest stable release should always be functional. If a game update renders the latest release functional, it doesn't make much sense to create a pre-release. Both Github and the EXILED Installer direct users towards downloading the latest release, which is now no longer functioning.

Even if a new update causes the latest pre-release to be potentially unstable, having a release with potential bugs versus a release that is completely non-functional seems to me like the obvious solution. Or, at the very least, versions shouldn't stay as pre-releases weeks after the SL update is complete there has been time to fix bugs.

Of course, versioning is another issue with this, as updates to the latest game version and small fixes shouldn't prompt entire new versions, but that's a whole different story.

tl;dr: The pre-release should be used for testing and game betas. A release should be used for the latest stable version. Because the latest release is almost always outdated, Github and the Installer will cause issues for people downloading EXILED. Users should not be forced to download a pre-release because the latest release is completely non-functional, and not just because of a bug. It doesn't make much sense to choose a non-functional latest release over one with bugs.
